editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/configurators/MavenTemplateConfigurator.java

```diff
@@ -44,13 +45,27 @@ public boolean configure(IProject project, CommonNewProjectMetaData metadata, IP
 		if (ok) {
 			// by default add the maven nature
 			ok = configureMavenNature(project, subMonitor.newChild(1));
+			new BuildAndRefreshJobWaiterUtil().waitJob(subMonitor.newChild(1));
 		}
-		
-		if (ok && !Strings.isBlank(metadata.getCamelVersion())) {
-			// by default configure the version of camel used in the pom.xml
-			ok = MavenUtils.configurePomCamelVersion(project, metadata, metadata.getCamelVersion(), subMonitor.newChild(1));
+
+		// to not trigger too many events while configuring the pom  we reuse the model
+		Model m2model = new CamelMavenUtils().getMavenModel(project);
+
+		try {
+			if (ok) {
+				// by default add staging repos if option enabled
+				ok = MavenUtils.manageStagingRepositories(m2model);
+			}
+			subMonitor.setWorkRemaining(2);
+			
+			if (ok && !Strings.isBlank(metadata.getCamelVersion())) {
```

Comment:
Now that the Camel version cannot be changed in UI, I think that when creating the metadata for project creation we should not add the camel version, it is automatically derived from the poms. The version 2.20.1 is always added currently.
It will avoid the "duplicating managed version" (and save time during creation)